addons/product/models/res_currency.py

```diff
@@ -11,5 +11,6 @@ def _activate_group_multi_currency(self):
         # for Sale/ POS - Multi currency flows require pricelists
         super()._activate_group_multi_currency()
         group_user = self.env.ref('base.group_user').sudo()
-        group_user._apply_group(self.env.ref('product.group_product_pricelist'))
-        self.env['res.company']._activate_or_create_pricelists()
+        if not group_user.user_has_groups('product.group_product_pricelist'):
```

Comment:
```suggestion
        if not self.user_has_groups('product.group_product_pricelist'):
```
confusing to call user_has_groups on the group_user when all it needs is the environment of the record, not the record itself.